### ISSUE TITLE
make setup failure

### ISSUE THREAD

**Issue by sachints123** [RedisTimeSeries/RedisTimeSeries#777]
Hi,
I am getting make setup error shown below on Ubuntu 18.04. Can you please suggest what is wrong?

```
:/home/sats/RedisTimeSeries# make setup
Setting up system...
apt-get -qq update -y
/usr/bin/python3  -m pip install --disable-pip-version-check wheel
/usr/bin/python3  -m pip install --disable-pip-version-check setuptools --upgrade
apt-get -qq install -y git jq curl
/home/sats/RedisTimeSeries/deps/readies/bin/enable-utf8
/home/sats/RedisTimeSeries/deps/readies/bin/getgcc
/usr/bin/python3  /home/sats/RedisTimeSeries/deps/readies/bin/getrmpytools
apt-get -qq update -y
/usr/bin/python3  -m pip install --disable-pip-version-check psutil
apt-get -qq install -y git
/usr/bin/python3  -m pip uninstall --disable-pip-version-check -y redis redis-py-cluster ramp-packer RLTest || true
/usr/bin/python3  -m pip install --disable-pip-version-check --no-cache-dir --ignore-installed redis-py-cluster
/usr/bin/python3  -m pip install --disable-pip-version-check --no-cache-dir --ignore-installed git+https://github.com/RedisLabsModules/RLTest.git@master
Collecting git+https://github.com/RedisLabsModules/RLTest.git@master
  Cloning https://github.com/RedisLabsModules/RLTest.git (to master) to /tmp/pip-1lsrc4z0-build
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/usr/lib/python3.6/tokenize.py", line 452, in open
        buffer = _builtin_open(filename, 'rb')
    FileNotFoundError: [Errno 2] No such file or directory: '/tmp/pip-1lsrc4z0-build/setup.py'
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-1lsrc4z0-build/
command failed: { /usr/bin/python3  -m pip install --disable-pip-version-check --no-cache-dir --ignore-installed git+https://github.com/RedisLabsModules/RLTest.git@master; } >/tmp/tmpgrp05vby 2>&1
command failed: { /usr/bin/python3  /home/sats/RedisTimeSeries/deps/readies/bin/getrmpytools; } >/tmp/tmpqvwsljvu 2>&1
Makefile:23: recipe for target 'setup' failed
make: *** [setup] Error 1
```

Thanks,
Sachin

**Comment by danni-m** [comment]
@chayim @rafie can you take a look please?

**Comment by chayim** [comment]
Looks like this was resolved on Ubuntu 18.04/bionic a while back.  Things were failing for you @sachints123 because you didn't have python3 **and** an up-to-date pip.  The getpy3 script (see below) does that.

Here's how I validated:

```
git clone --recursive git@github.com:redistimeseries/redistimeseries
docker run -v `pwd`:/build -it ubuntu:bionic
```

#### now, in the docker
```
cd /build
apt-get update
./deps/readies/bin/getpy3
make setup
```

The issue is that older versions of pip don't look for pyproject.toml ([PEP 508](https://www.python.org/dev/peps/pep-0508/)) files in a repo, and require setup.py. [rltest](https://github.com/RedisLabsModules/RLTest) dependency used for testing was failing to install due to this issue.
